src/app/submissions/submissions.component.ts

```diff
@@ -50,7 +50,8 @@ export class SubmissionsComponent implements OnInit, AfterViewInit, OnDestroy {
     this.submissionsService.submissionsUpdated$.pipe(takeUntil(this.onDestroy$))
     .subscribe((submissions) => {
       this.submissions.data = submissions.map(submission => ({
-        ...submission, submittedBy: submission.user.name || (submission.user.firstName + ' ' + submission.user.lastName).trim()
+        ...submission, submittedBy: submission.user.name ?
+        submission.user.name || (submission.user.firstName + ' ' + submission.user.lastName).trim() : ''
```

Comment:
@singharpita This will never result in `(submission.user.firstName + ' ' + submission.user.lastName).trim()`.  If `submission.user.name` is truthy/has a value, it will always end up as `submission.user.name`.  If it is not it will be an empty string.

We need a solution which uses the `name` property of the user if it exists, the `firstName` + `lastName` if it does not, and "Unknown" if it neither exist.

Reply:
oops @paulbert both of us commented :D